src/axom/klee/io/IO.cpp

```diff
@@ -217,20 +225,25 @@ Geometry convert(GeometryData const &data,
                      data.path,
                      data.operatorData.makeOperator(startProperties, namedOperators)};
 
-  if(geometry.getEndProperties().dimensions != fileDimensions)
+  const auto computed_end_dims = geometry.getEndProperties().dimensions;
+  const auto expected_end_dims = has_explicit_dims ? data.explicitDimensions : fileDimensions;
+  if(computed_end_dims != expected_end_dims)
   {
-    throw KleeError(
-      {data.pathInFile, "Did not end up in the number of dimensions specified by the file"});
+    throw KleeError({data.pathInFile,
+                     axom::fmt::format("Did not end up with the expected number of dimensions. "
+                                       "Expected: {}, but got: {}",
+                                       expected_end_dims == Dimensions::Two ? 2 : 3,
```

Comment:
Is there any way the dimensions could still be `Unspecified`? Would an fmt formatter be appropriate?

Reply:
Great suggestion!

I added a custom formatter for the enum.

In the process, I realized that `klee::Dimensions dim {}` would get a value of `0`, which would not match the current value for `Dimensions::Unspecified /* ==-1 */`, so I modified the value of `Dimensions::Unspecified` to be `0` instead of `-1`